### PR TITLE
Resolve #223

### LINKED ISSUE #223: [FEATURE] Add command to move events out of the archive

### PR DESCRIPTION
**Pull request by Denio21**
update

### CODE REVIEW

File: src/commands/EventCommand.js [hunk 1]
```diff
@@ -378,6 +381,85 @@ class EventCommand extends SubsCommand {
...
+                    return moment(values["start"], "D. M. YYYY HH:mm").add(45, "m").format("D. M. YYYY HH:mm");
+
+                else
+                    return content;
+            }
+        },
+
+        ], (values) => {
+            logger.info("User " + message.member.displayName + " unarchived event with name " + values["name"] + ".");
+
```
Comment (cajthamlmatej) [inline src/commands/EventCommand.js]: You dont have placeholders here, you cant have pure placeholders in DB!

File: src/commands/EventCommand.js [hunk 2]
```diff
@@ -378,6 +381,85 @@ class EventCommand extends SubsCommand {
...
+                if (!(moment(content, "D. M. YYYY").isValid() || moment(content, "D. M. YYYY HH:mm").isValid()))
+                    return "command.event.wrong-date-format";
+                else
+                    return true;
+            },
+            "value": (content, values) => {
+                if (content == "-")
+                    return values["start"];
+
+                else if (content == "+45")
```
Comment (cajthamlmatej) [inline src/commands/EventCommand.js]: This should be here anywhere!

File: src/database/repositories/EventRepository.js [hunk 3]
```diff
@@ -129,6 +134,23 @@ class EventRepository extends Repository {
             start: { $regex: "^" + day }
         });
     }
+    async getArchivedEvents(archived = true, fields = null) {
```
Comment (cajthamlmatej) [inline src/database/repositories/EventRepository.js]: You dont have to have this method here, you can use getEvents(archived = false, fields = null) and use acrhived = true...

File: src/commands/EventCommand.js [hunk 4]
```diff
@@ -175,7 +178,7 @@ class EventCommand extends SubsCommand {
                 if (content == "-")
                     return values["start"];
 
-                else if (content == "+45") 
+                else if (content == "+45")
```
Comment (cajthamlmatej) [inline src/commands/EventCommand.js]: This should not be here...

**Comment by cajthamlmatej** [comment]
And also you didnt have added what I requested. You have to check if unarchived events exists in event creation and dismiss it.

File: src/modules/EventModule.js [hunk 5]
```diff
@@ -203,6 +219,10 @@ class EventModule extends Module {
         return await eventRepository.doesEventExistsWithName(name);
     }
 
+    async archiveexists(name) {
```
Comment (cajthamlmatej) [inline src/modules/EventModule.js]: archiveExists!!!!!!!!!!!!!!!!!!!!!!!

File: src/database/repositories/EventRepository.js [hunk 6]
```diff
@@ -11,11 +11,11 @@ class EventRepository extends Repository {
     }
 
     async doesEventExistsWithName(name) {
-        const entity = await this.entity.findOne({ name: name, archived: false, deleted: false });
+        const entity = await this.entity.findOne({ name: name, deleted: false });
         return entity != null;
     }
 
-    async getEvents(archived = false, fields = null) {
+    async getEvents(archived = true, fields = null) {
```
Comment (cajthamlmatej) [inline src/database/repositories/EventRepository.js]: archived is default parameter!!!!!! use archived = false!!!!!!!!!!!!!!

File: src/modules/EventModule.js [hunk 7]
```diff
@@ -224,6 +244,27 @@ class EventModule extends Module {
         user.createDM().then(dm => dm.send(embed)).catch(logger.error);
     }
 
+    async printArchivedList(user) {
+        let list = "";
+        const events = await eventRepository.getArchivedEvents(true, "name title");
```
Comment (cajthamlmatej) [inline src/modules/EventModule.js]: This method doesnt even exists!!! Use getEvents(true, ...)

File: src/database/repositories/EventRepository.js (hunk 6 above)
Comment (Denio21) [reply src/database/repositories/EventRepository.js]: sxjbcv klmansc¨§

File: src/database/repositories/EventRepository.js (hunk 3 above)
Comment (Denio21) [reply src/database/repositories/EventRepository.js]: :)))))

File: src/commands/EventCommand.js [hunk 8]
```diff
@@ -378,6 +378,93 @@ class EventCommand extends SubsCommand {
...
+
+                if (!(moment(content, "D. M. YYYY").isValid() || moment(content, "D. M. YYYY HH:mm").isValid()))
+                    return "command.event.wrong-date-format";
+                else
+                    return true;
+            }
+        },
+        {
+            "name": "end",
+            "example": ["-", "+45", moment().add(3, "days").format("D. M. YYYY"), moment().add(3, "days").format("D. M. YYYY HH:mm"), ... Object.keys(this.placeholders)],
```
Comment (cajthamlmatej) [inline src/commands/EventCommand.js]: remove +45

File: src/commands/EventCommand.js [hunk 9]
```diff
@@ -378,6 +378,93 @@ class EventCommand extends SubsCommand {
...
+            "commands": [{
+                reaction: "➖",
+                value: "-"
+            }],
+            "validate": (content) => {
+                if (content == "-" || content == "+45")
```
Comment (cajthamlmatej) [inline src/commands/EventCommand.js]: hmmmmm

File: src/commands/EventCommand.js (hunk 8 above)
Comment (Denio21) [reply src/commands/EventCommand.js]: l.m,nůl

>

**Pull request merged**